libkbfs/mdserver_memory.go

```diff
@@ -371,8 +371,11 @@ func (md *MDServerMemory) Put(ctx context.Context, rmds *RootMetadataSigned) err
 
 	// Consistency checks
 	if head != nil {
-		err := head.MD.CheckValidSuccessorForServer(
-			md.config.Crypto(), &rmds.MD)
+		id, err := md.config.Crypto().MakeMdID(&head.MD)
```

Comment:
Oh, are we concerned about the local-server KBFS case? I had thought that that was free to make unoptimized.

I don't think this change actually adds many additional MdID calls on the client-side -- we still calculate the MdID exactly once for each retrieved or created MD, I believe, and even previously we needed the MetadataID for checking PrevRoot anyway. But I can run some perf tests.